src/main/java/com/google/sps/servlets/NewRecipeServlet.java

```diff
@@ -27,9 +17,39 @@
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // implement new recipe
+    String recipeName = request.getParameter("recipeName");
+    String tagsResponse = request.getParameter("tags");
+    String ingredientsResponse = request.getParameter("ingredients");
+    String stepsResponse = request.getParameter("steps");
+    String privacy = request.getParameter("privacy");
 
-    response.sendRedirect("/pages/UserPage.html");
-  }
+    List<String> tags = new ArrayList<String>(Arrays.asList(tagsResponse.split(",", 0)));
+    List<String> ingredients = new ArrayList<String>(Arrays.asList(ingredientsResponse.split("\n", 0)));
+    List<String> steps = new ArrayList<String>(Arrays.asList(stepsResponse.split("\n", 0)));
+
+    tags.removeAll(Arrays.asList("", null));
+    ingredients.removeAll(Arrays.asList("", null, "\n", "\r\n", "\r"));
+    steps.removeAll(Arrays.asList("", null, "\n", "\r\n", "\r"));
+
+    for (String tag : tags) {
+      tag.trim();
+    }
+    for (String ingredient : ingredients) {
+      ingredient.trim();
+    }
+    for (String step : steps) {
+      step.trim();
+    }
 
-}
+    Entity recipeEntity = new Entity("PrivateRecipe");
```

Comment:
Will all recipes be saved as this type of entity later, or will you include an if-statement later to control the type of entity creation?

Reply:
later when we have public recipes I'll have an if statement